packages/components/src/components/as-category-widget/as-category-widget.tsx

```diff
@@ -0,0 +1,210 @@
+      <p class='as-category-widget__description as-body'>{this.description}</p>,
+    ];
+  }
+
+  private _renderCategoryList() {
+    return <ul class='as-category-widget__list'>{this._renderCategories()}</ul>;
+  }
+
+  private _renderCategories() {
+    const moreCategoriesThanVisible = this.categories.length > this.visibleCategories;
+    const { categories, otherCategory } = this._parseCategories();
+    let otherCategoryTemplate;
+
+    const categoriesToRender =  moreCategoriesThanVisible
+      ? categories.slice(0, this.visibleCategories)
+      : categories;
+
+    const maximumValue = this.useTotalPercentage
+      ? this._getCategoriesTotalValue(this.categories)
+      : this._getVisibleCategoriesMaximumValue();
+
+    if (otherCategory || moreCategoriesThanVisible) {
+      otherCategoryTemplate = this._renderOtherCategory(otherCategory, { maximumValue });
+    }
+
+    return [
+      categoriesToRender.map((category: Category) => this._renderCategory(category, { maximumValue })),
+      otherCategoryTemplate
+    ];
+  }
+
+  private _renderCategory(category: Category, options: CategoryOptions) {
+    const isSelected = this._isSelected(category.name);
+    const isAnyCategorySelected = this.selectedCategories.length > 0;
+    const barColor = options.isOther
+      ? OTHER_CATEGORY_COLOR
+      : this._getBarColor(category.color || this.defaultBarColor, { isSelected });
+
+    const progressStyles = {
+      backgroundColor: barColor,
+      width: `${(category.value / options.maximumValue) * 100}%`
+    };
+
+    const cssClasses = {
+      'as-category-widget__category': true,
+      'as-category-widget__category--not-selected': isAnyCategorySelected && (!isSelected || options.isOther),
+      'as-category-widget__category--other': options.isOther,
+      'as-category-widget__category--selected': isSelected
+    };
+
+    return (
+      <li class={cssClasses} onClick={() => this._toggleCategory(category)}>
+        <p class='as-category-widget__info as-body'>
+          <div class='as-category-widget__title'>{category.name}</div>
+          {readableNumber(category.value)}
+        </p>
+
+        <div class='as-category-widget__bar'>
+          <div class='as-category-widget__bar-value' style={progressStyles}></div>
+        </div>
+      </li>
+    );
+  }
+
+  private _renderOtherCategory(category: Category, options: CategoryOptions) {
+    const categoryData = category || {
+      name: 'Other',
+      value: this._getCategoriesTotalValue(
+        this.categories.slice(this.visibleCategories, this.categories.length)
+      )
+    };
+
+    return this._renderCategory(categoryData, { maximumValue: options.maximumValue, isOther: true });
+  }
+
+  private _renderFooter() {
+    const selectedCount = this.selectedCategories.length || 'All';
+
+    return (
+      <footer class='as-category-widget__footer'>
+        <div class='as-category-widget__count as-body'>{selectedCount} selected</div>
+        { this.showClearButton
```

Comment:
What I use to do in react was:

```
{ this.showClearButton && <button class='as-category-widget__clear' onClick={() => this.clearSelection()}>Clear selection</button> }
```

I think it looks cleaner, but it's just my opinion 😂